lib/core/collections.toit

```diff
@@ -3544,15 +3584,20 @@ class Deque extends List implements Collection:
   */
   reserve amount/int:
     if amount < 0: throw "OUT_OF_BOUNDS"
-    backing_.resize (backing_.size + amount)
+    if backing_.size - last_ >= amount: return
+    backing_.resize (last_ + amount)
 
   shrink-if-needed_ -> none:
     backing := backing_
     first := first_
-    if first * 2 > backing.size:
-      backing.replace 0 backing first first + size
-      backing.resize size
+    last := last_
+    reserved := backing.size - last
+    sz := last - first
+    if first * 2 > last:
```

Comment:
Add a comment here? Why do we multiply here and shift in the other place?

Reply:
switched to shift for consistency.
done.